### PR TITLE
follow-up #17930 - inline syntax highlighting

### PR DESCRIPTION
**Pull request by a-mr**
Improves syntax highlighting in `gc.rst`.

Fixes a bug that `option` could not be set as default-role.

Proper displaying of option list requires #17845.

cc @narimiran

### CODE REVIEW

**Review by Varriount — APPROVED** [review]

File: lib/packages/docutils/rst.nim [hunk 1]
```diff
@@ -1151,6 +1151,17 @@ proc toOtherRole(n: PRstNode, kind: RstNodeKind, roleName: string): PRstNode =
...
   var newKind = n.kind
   var newSons = n.sons
+
+  proc finalizeInterpreted(node: PRstNode, newKind: RstNodeKind,
+                           roleName: string): PRstNode =
+    # fixes interpreted text (`x` or `y`:role:) to proper internal AST format
+    if newKind in {rnUnknownRole, rnCodeFragment}:
+      result = n.toOtherRole(newKind, roleName)
+    elif newKind == rnInlineCode:
+      result = n.toInlineCode(language=roleName)
```
Comment (Araq) [inline lib/packages/docutils/rst.nim]: `n` vs `node`

File: lib/packages/docutils/rst.nim [hunk 2]
```diff
@@ -1151,6 +1151,17 @@ proc toOtherRole(n: PRstNode, kind: RstNodeKind, roleName: string): PRstNode =
 proc parsePostfix(p: var RstParser, n: PRstNode): PRstNode =
   var newKind = n.kind
   var newSons = n.sons
+
+  proc finalizeInterpreted(node: PRstNode, newKind: RstNodeKind,
+                           roleName: string): PRstNode =
+    # fixes interpreted text (`x` or `y`:role:) to proper internal AST format
+    if newKind in {rnUnknownRole, rnCodeFragment}:
+      result = n.toOtherRole(newKind, roleName)
```
Comment (Araq) [inline lib/packages/docutils/rst.nim]: `n` vs `node`

File: lib/packages/docutils/rst.nim [hunk 3]
```diff
@@ -1151,6 +1151,17 @@ proc toOtherRole(n: PRstNode, kind: RstNodeKind, roleName: string): PRstNode =
 proc parsePostfix(p: var RstParser, n: PRstNode): PRstNode =
   var newKind = n.kind
   var newSons = n.sons
+
+  proc finalizeInterpreted(node: PRstNode, newKind: RstNodeKind,
+                           roleName: string): PRstNode =
```
Comment (Araq) [inline lib/packages/docutils/rst.nim]: Use `.nimcall` to prevent the closure overhead.

Comment (a-mr) [reply lib/packages/docutils/rst.nim]: thanks, changed

**Pull request merged**